src/explorer/SiteTreeItem.ts

```diff
@@ -85,43 +84,14 @@ export abstract class SiteTreeItem implements IAzureParentTreeItem {
         await this.siteWrapper.enableHttpLogs(client);
     }
 
-    public async connectToLogStream(client: WebSiteManagementClient, extensionContext: ExtensionContext): Promise<OutputChannel> {
-        const siteName = this.siteWrapper.appName;
-        const user = await util.getWebAppPublishCredential(client, this.site);
-        const kuduClient = new KuduClient(siteName, user.publishingUserName, user.publishingPassword);
-
-        if (!this._logStreamOutputChannel) {
-            this._logStreamOutputChannel = window.createOutputChannel(`${siteName} - Log Stream`);
-            extensionContext.subscriptions.push(this._logStreamOutputChannel);
-        }
-
-        this.stopLogStream();
-        this._logStreamOutputChannel.appendLine('Connecting to log-streaming service...');
-        this._logStreamOutputChannel.show();
-
-        this._logStream = kuduClient.getLogStream().on('data', chunk => {
-            this._logStreamOutputChannel.append(chunk.toString());
-        }).on('error', err => {
-            util.sendTelemetry('ConnectToLogStreamError', { name: err.name, message: err.message });
-            this._logStreamOutputChannel.appendLine('Error connecting to log-streaming service:');
-            this._logStreamOutputChannel.appendLine(err.message);
-        }).on('complete', () => {
-            this._logStreamOutputChannel.appendLine('Disconnected from log-streaming service.');
-        });
-
-        return this._logStreamOutputChannel;
-    }
-
-    public stopLogStream(): void {
-        if (this._logStream) {
-            this._logStream.removeAllListeners();
-            this._logStream.destroy();
-            this._logStream = undefined;
-
-            if (this._logStreamOutputChannel) {
-                this._logStreamOutputChannel.appendLine('Disconnected from log-streaming service.');
-            }
+    public async connectToLogStream(client: WebSiteManagementClient, actionHandler: AzureActionHandler, context: ExtensionContext): Promise<ILogStream> {
+        const kuduClient: KuduClient = await this.siteWrapper.getKuduClient(client);
+        if (!this.logStreamOutputChannel) {
+            const logStreamoutputChannel: OutputChannel = window.createOutputChannel(`${this.label} - Log Stream`);
```

Comment:
I prefer `this.siteWrapper.appName` over `this.label` (which might have 'Stopped' in the name)